operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go

```diff
@@ -135,14 +135,19 @@ func checkTraffic(spec *SeldonDeploymentSpec, fldPath *field.Path, allErrs field
 			if shadows > 1 {
 				allErrs = append(allErrs, field.Invalid(fldPath, spec.Predictors[i].Name, "Multiple shadows are not allowed"))
 			}
+			if p.Traffic < 0 || p.Traffic > 100 {
+				allErrs = append(allErrs, field.Invalid(fldPath, spec.Predictors[i].Name, "shadow traffic is illegal, the traffic number should be between [0, 100]"))
+			}
 		}
 	}
 
-	if trafficSum != 100 && (len(spec.Predictors)-shadows) > 1 {
-		allErrs = append(allErrs, field.Invalid(fldPath, spec.Predictors[0].Name, "Traffic must sum to 100 for multiple predictors"))
-	}
-	if trafficSum > 0 && trafficSum < 100 && len(spec.Predictors) == 1 {
-		allErrs = append(allErrs, field.Invalid(fldPath, spec.Predictors[0].Name, "Traffic must sum be 100 for a single predictor when set"))
+	if shadows == 0 {
```

Comment:
I think we need to do the checks even if shadows are present. We just need to ignore the shadow percentage in the calculation.